cli/src/program.rs

```diff
@@ -2627,6 +2632,70 @@ fn check_payer(
     Ok(())
 }
 
+// This enum is equivalent to an Option but was added to self-document
+// the ok variants and has the benefit of not forcing the caller to use
+// the result if they don't care about it.
+enum UpdateComputeUnitLimitResult {
+    UpdatedInstructionIndex(usize),
+    NoInstructionFound,
+}
+
+// Returns the index of the compute unit limit instruction
+fn simulate_and_update_compute_unit_limit(
+    rpc_client: &RpcClient,
+    transaction: &mut Transaction,
+) -> Result<UpdateComputeUnitLimitResult, Box<dyn std::error::Error>> {
+    let Some(compute_unit_limit_ix_index) = transaction
+        .message
+        .instructions
+        .iter()
+        .enumerate()
+        .filter_map(|(ix_index, instruction)| {
+            let ix_program_id = transaction.message.program_id(ix_index)?;
+            if ix_program_id != &compute_budget::id() {
+                return None;
+            }
+
+            match try_from_slice_unchecked(&instruction.data).ok()? {
+                ComputeBudgetInstruction::SetComputeUnitLimit(_) => Some(ix_index),
+                _ => None,
+            }
```

Comment:
nit: I got a little confused by the `ok()?` *and also* the `None` in the match, so how about using `std::matches` here? ie

```
matches!(
    try_from_slice_unchecked(&instruction.data),
    Ok(ComputeBudgetInstruction::SetComputeUnitLimit(_))
).then_some(ix_index)
```